src/polymetrix/featurizers/polymer.py

```diff
@@ -45,42 +46,56 @@ def from_psmiles(cls, psmiles: str) -> "Polymer":
 
     @property
     def psmiles(self) -> Optional[str]:
-        """Gets the pSMILES string of the polymer.
-
-        Returns:
-            Optional[str]: The pSMILES string, or None if not set.
-        """
+        """The pSMILES string of the polymer."""
         return self._psmiles
 
     @psmiles.setter
     def psmiles(self, value: str):
-        """Sets the pSMILES string and updates the polymer's internal structure.
+        """Sets the pSMILES string and updates the polymer's structure.
 
         Args:
-            value: str, the pSMILES string to set.
+            value: The pSMILES string to set.
 
         Raises:
-            ValueError: If the pSMILES string is invalid or cannot be processed.
+            ValueError: If the pSMILES string is None, empty, or invalid.
         """
+        if not value or not isinstance(value, str):
+            raise ValueError("pSMILES cannot be None or empty")
         try:
             mol = Chem.MolFromSmiles(value)
             if mol is None:
                 raise ValueError("Invalid pSMILES string")
             self._psmiles = value
-            self._graph = self._mol_to_nx(mol)
+            self._mol = mol
+            self.graph = self._mol_to_nx(mol)
             self._identify_connection_points()
             self._identify_backbone_and_sidechain()
         except Exception as e:
             raise ValueError(f"Error processing pSMILES: {str(e)}") from e
 
-    def _mol_to_nx(self, mol: Chem.Mol) -> nx.Graph:
+    @property
+    def terminal_groups(self) -> Optional[Dict[int, str]]:
```

Comment:
this is now the hack I mentioned. But you should perhaps also think about how you want to integrate this long term. Do you want to have some of this as part of the molecule object?

Reply:
i.e. should the molecule object - in the long term - only be the monomer. Or should it be the polymer that thus also contains info about DOP and terminal groups?